src/Text/Pandoc/Writers/RST.hs

```diff
@@ -918,25 +918,72 @@ imageDimsToRST attr = do
 simpleTable :: PandocMonad m
             => WriterOptions
             -> (WriterOptions -> [Block] -> m (Doc Text))
-            -> [[Block]]
-            -> [[[Block]]]
+            -> TableHead
+            -> [TableBody]
+            -> TableFoot
             -> m (Doc Text)
-simpleTable opts blocksToDoc headers rows = do
-  -- can't have empty cells in first column:
-  let fixEmpties (d:ds) = if isEmpty d
-                             then literal "\\ " : ds
-                             else d : ds
-      fixEmpties [] = []
-  headerDocs <- if all null headers
+simpleTable opts blocksToDoc (TableHead _ headers) tbody (TableFoot _ footers) = do
+  headerDocs <- if null headers
                    then return []
-                   else fixEmpties <$> mapM (blocksToDoc opts) headers
-  rowDocs <- mapM (fmap fixEmpties . mapM (blocksToDoc opts)) rows
-  let numChars = maybe 0 maximum . NE.nonEmpty . map offset
-  let colWidths = map numChars $ transpose (headerDocs : rowDocs)
-  let toRow = mconcat . intersperse (lblock 1 " ") . zipWith lblock colWidths
+                   else fixEmpties <$> mapM rowToDoc headers
+  rowDocs <- fixEmpties <$> mapM rowToDoc ((concatMap tableBodyToRows tbody) ++ footers)
+  let numChars = maybe 0 maximum . NE.nonEmpty . map (offset . fst)
+  let colWidths = map numChars $ transpose (headerDocs ++ rowDocs)
   let hline = nowrap $ hsep (map (\n -> literal (T.replicate n "=")) colWidths)
-  let hdr = if all null headers
+  let hdr = if null headers
```

Comment:
Why the change from `all null` to `null`?  This could make a difference.

Reply:
Still, maybe we need to also check for the case where the Rows aren't empty but they contain only empty cells?  That was the case we were catching before.

Reply:
I'm not absolutely sure it's necessary, but I worry that it will be a regression.

Reply:
Gotcha. I'm gonna try to adjust my code to ensure that this check will still hold.